tasks/quality_gates.py

```diff
@@ -295,6 +325,48 @@ def manual_threshold_update(self, filename="static_gate_report.json"):
     notify_threshold_update(pr_url)
 
 
+@task
+def debug_specific_quality_gate(ctx, gate_name):
+    """
+    Executes a single static quality gate to compare it to its ancestor and run debug on it
+
+    :param ctx: Invoke context
+    :param gate_name: Static quality gates to debug
+    :return:
+    """
+    if not gate_name:
+        raise Exit(
+            code=0,
+            message="Please ensure to set the GATE_NAME variable inside of the manual job execution gitlab page when executing this debug job.",
+        )
+    nightly_run = False
+    branch = os.environ["CI_COMMIT_BRANCH"]
+
+    DDR_WORKFLOW_ID = os.environ.get("DDR_WORKFLOW_ID")
+    if DDR_WORKFLOW_ID and branch == "main" and is_conductor_scheduled_pipeline():
+        nightly_run = True
+
+    quality_gates_module = __import__("tasks.static_quality_gates", fromlist=[gate_name])
+    gate_inputs = {"ctx": ctx, "nightly": nightly_run}
+    try:
+        gate_module = getattr(quality_gates_module, gate_name)
+    except AttributeError as e:
+        raise Exit(
+            code=0,
+            message=f"The provided quality gate to debug ({gate_name}) is invalid and wasn't found as part of tasks.static_quality_gates.",
+        ) from e
+
+    # As it is a debug job we do not want the job to actually fail on failures.
```

Comment:
Given that this seems to be mainly intended to be run when gates have already failed, and that it's set to allow to fail, why not let it fail?